File: spring-boot-store-demo/src/main/java/com/linkedinlearning/springbootstoredemo/service/ProductService.java
package com.linkedinlearning.springbootstoredemo.service;

import com.linkedinlearning.springbootstoredemo.model.Product;
import org.jetbrains.annotations.NotNull;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.BeanPropertyRowMapper;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Encapsulates supported database interactions to be used by application controllers
 */
@Service
public class ProductService {

    @Autowired JdbcTemplate jdbcTemplate;

    public List<Product> allProducts() {
        String sql = "SELECT * FROM products";
        return jdbcTemplate.query(sql, BeanPropertyRowMapper.newInstance(Product.class));
    }

    @NotNull
    public Product insert(Product product) {
        jdbcTemplate.update(
                "INSERT INTO products VALUES(?, ?, ?, ?, ?)",
                product.getUpc(),
                product.getName(),
                product.getType(),
                product.getDescription(),
                product.getThumbnail()
        );
        return product;
    }
}

File: spring-boot-store-demo/src/main/java/com/linkedinlearning/springbootstoredemo/service/ProductService.kt
package com.linkedinlearning.springbootstoredemo.service

import com.linkedinlearning.springbootstoredemo.model.Product
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.jdbc.core.BeanPropertyRowMapper
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Service

/**
 * Encapsulates supported database interactions to be used by application controllers
 */
@Service
class ProductService {

    @Autowired
    lateinit var jdbcTemplate: JdbcTemplate

    fun allProducts(): List<Product> {
        val sql = "SELECT * FROM products"
        return jdbcTemplate.query(sql, BeanPropertyRowMapper.newInstance(Product::class.java))
    }

    fun insert(product: Product): Product {
        jdbcTemplate.update(
            "INSERT INTO products VALUES(?, ?, ?, ?, ?)",
            product.upc,
            product.name,
            product.type,
            product.description,
            product.thumbnail
        )
        return product
    }
}